abd-clam/src/core/cluster/tree.rs

```diff
@@ -72,4 +78,372 @@ impl<I: Instance, U: Number, D: Dataset<I, U>> Tree<I, U, D> {
     pub const fn radius(&self) -> U {
         self.root.radius
     }
+
+    /// Saves a tree to a given location
+    ///
+    /// The path given will point to a newly created folder which will
+    /// store all necessary data for tree reconstruction.
+    ///
+    /// The directory structure looks like the following:
+    /// ```text
+    /// user_given_path/
+    ///      clusters/   <-- Clusters are serialized using their hex name.
+    ///      childinfo/  <-- Information about clusters immediate children.
+    ///      dataset/    <-- The serialized dataset.
+    ///      leaves.json <-- A json file of leaf names.
+    /// ````
+    ///
+    /// # Errors
+    /// Errors out on any directory or file creation issues
+    #[allow(clippy::missing_panics_doc)]
+    pub fn save(&self, path: &Path) -> Result<(), String> {
+        // Create our directory if needed
+        let dirbuilder = DirBuilder::new();
```

Comment:
What is the difference between `DirBuilder::create` and `File::create`?

Reply:
Couldn't find a reason to use `DirBuilder` directly, but it seems like `std::fs::create_dir` is the standard way to create directories.